common/router_data_model.go

```diff
@@ -561,50 +1105,136 @@ func (rdm *RouterDataModel) HandleConfigTypeEvent(index uint64, event *edge_ctrl
 // during startup.
 func (rdm *RouterDataModel) HandleConfigEvent(index uint64, event *edge_ctrl_pb.DataState_Event, model *edge_ctrl_pb.DataState_Event_Config) {
 	if event.Action == edge_ctrl_pb.DataState_Delete {
-		rdm.Configs.Remove(model.Config.Id)
+		rdm.Configs.RemoveCb(model.Config.Id, func(key string, v *Config, exists bool) bool {
+			if v != nil {
+				if !v.identities.IsEmpty() {
+					panic(fmt.Errorf("config %s still has identity references", key))
+				}
+				if !v.services.IsEmpty() {
+					panic(fmt.Errorf("config %s still has service references", key))
```

Comment:
I want to confrim the intention here:

1) crash the router
2) not use pfxlog.Logger().Fataf if so

Reply:
removed the panics. I wanted fast fail during tests, but now that the test is running clean, we shouldn't hit this. Maybe I need to add a test mode